squad/core/tasks/notification.py

```diff
@@ -7,14 +9,40 @@
 import logging
 
 
-@celery.task(bind=True)
-def notify_project_status(self, status_id):
-    try:
-        status = ProjectStatus.objects.get(pk=status_id)
-        send_status_notification(status)
-    except ProjectStatus.DoesNotExist as not_found:
-        last_id = ProjectStatus.objects.all().aggregate(Max('id'))['id__max']
-        if not last_id or status_id > last_id:
-            # our id is larger than the latest, the object was probably created
-            # by another process but not commited to the database yet.
-            raise self.retry(exc=not_found, countdown=30)  # retry in 30s
+@celery.task
+def maybe_notify_project_status(status_id):
+    projectstatus = ProjectStatus.objects.get(pk=status_id)
+    build = projectstatus.build
+    project = build.project
+
+    timeout = project.notification_timeout
+    if timeout:
+        notification_timeout.apply_async(args=[status_id], countdown=timeout)
```

Comment:
(reposted comment because i originally posted within the commit and github thinks it's 'outdated')

This guarantees a notification but only if maybe_notify_project_status has been called, but doesn't that require at least one result? What if no results have been received?

Same question about concurrency - could this result in duplicate emails if multiple notification_timeouts are queued up for the same moment?